app/src/ticketForm.js

```diff
@@ -0,0 +1,184 @@
+import { version } from '../package.json';
+import algoliasearch from 'algoliasearch/lite';
+import { highlight, snippet } from 'instantsearch.js/es/helpers';
+import './ticketform.css';
+// eslint-disable-next-line no-unused-vars
+import { render, h, Fragment } from 'preact';
+import { useState } from 'preact/hooks';
+import translate from './translations';
+
+import { debounceGetAnswers } from './answers';
+import { initInsights, extendWithConversionTracking } from './clickAnalytics';
+
+class TicketForm {
+  constructor({
+    applicationId,
+    apiKey,
+    ticketForms: { enabled },
+    indexPrefix,
+    subdomain,
+    clickAnalytics,
+  }) {
+    if (!enabled) return;
+    this.client = algoliasearch(applicationId, apiKey);
+    this.client.addAlgoliaAgent(`Zendesk Integration (${version})`);
+    this.indexName = `${indexPrefix}${subdomain}_articles`;
+
+    if (clickAnalytics) {
+      initInsights({ applicationId, apiKey });
+      extendWithConversionTracking(this, {
+        clickAnalytics,
+        indexName: this.indexName,
+      });
+    }
+  }
+
+  init({
+    baseUrl,
+    clickAnalytics,
+    locale,
+    translations,
+    ticketForms: {
+      enabled,
+      inputSelector,
+      suggestionsListSelector,
+      descriptionSelector,
+      answersParameters,
+      requireSubject,
+      cssClasses: {
+        descriptionGroup,
+        disabledDescriptionGroup,
+        descriptionWarning,
+        suggestionsList,
+      },
+    },
+  }) {
+    if (!enabled) return;
+
+    // eslint-disable-next-line consistent-this
+    const self = this;
+
+    if (requireSubject) {
+      self.descriptionElement = document.querySelector(descriptionSelector);
+      self.descriptionElement.classList.add(descriptionGroup);
+      self.descriptionElement.classList.add(disabledDescriptionGroup);
+      const warning = document.createElement('span');
+      warning.classList.add(descriptionWarning);
+      warning.textContent = translate(translations, locale, 'descriptionLock');
+      self.descriptionElement.append(warning);
+    }
+
+    document
+      .querySelector(suggestionsListSelector)
+      .setAttribute('style', 'display: none!important');
+
+    const lang = locale.split('-')[0];
+    const buildUrl = (hit) => `${baseUrl}${locale}/articles/${hit.id}`;
```

Comment:
Let's probably move `buildUrl` to a helper/util file, as it's used in 2 files now.